internal/alb/tg/targetgroup.go

```diff
@@ -150,6 +156,33 @@ func (controller *defaultController) reconcileTGInstance(ctx context.Context, in
 	return instance, nil
 }
 
+// resolveServiceHealthCheckPort checks if the service-port annotation is a string. If so, it tries to look up a port with the same name
+// on the service and use that port's NodePort as the health check port.
+func resolveServiceHealthCheckPort(store store.Storer, namespace string, serviceName string, servicePortAnnotation intstr.IntOrString) (* string) {
+	servicePort := servicePortAnnotation.String()
+
+	//If the annotation uses the default port ("traffic-port"), do not try to look up a port by that name.
+	if servicePortAnnotation.Type == intstr.String && servicePortAnnotation.StrVal != healthcheck.DefaultPort {
+		serviceKey := namespace + "/" + serviceName
+
+		service, err := store.GetService(serviceKey)
+		if err != nil {
+			return &servicePort
+		}
+
+		for _, p := range service.Spec.Ports {
+			if p.Name == servicePortAnnotation.StrVal && p.NodePort != 0 {
```

Comment:
this only works under instance mode, for ip mode(which targets pods directly), it should be the`targetPort`